bin/fpdiff.py

```diff
@@ -80,6 +80,10 @@ def fpdiff_helper(filename1,filename2,relative_error,small,
  # handle the error and so we shouldn't do anything weird here).
  tmpfile1 = read_file(filename1)
  tmpfile2 = read_file(filename2)
+ # this line catches the error when an empty selftest gold data file is 
+ # provided and will return a FAIL (e.g. wrong filename is passed for the selftest, etc.)
+ if len(tmpfile1) == 0:
+  return 1
```

Comment:
I think it would be helpful to provide a meaningful error message rather than silently exiting. Please update accordingly

Reply:
Agree that it would be useful to output that both files have zero length, which is treated as a fail.

Reply:
that makes sense, newest added an error message to the details_stream